src/Language/PureScript/TypeChecker/Kinds.hs

```diff
@@ -409,17 +410,41 @@ checkTypeKind
   -> SourceType
   -> m ()
 checkTypeKind ty kind =
-  unifyKindsWithFailure (\_ _ -> throwError . errorMessage $ ExpectedType ty kind) kind E.kindType
+  -- In this case, if any failure occurs, we ignore it and throw our own error.
+  -- So, rather than use `unifyishRows` here and pay for its overhead,
+  -- we'll use the approach that was used before we migrated to `unifyishRows`
+  -- because the original approach fails faster.
+  unifyKindsWithFailure (UseSequential ty kind) kind E.kindType
```

Comment:
While this should be faster due to the lack of overhead (unless there's GHC optimizations that will end up removing that), it also diverges from `unifyishRows`, which may cause issues in the future. Is that tradeoff worth it?

Reply:
I don't know about the performance impact or the potential issues caused by changing the implementation in subtle ways. I do however think that it's to our long-term benefit to maintain as few distinct code paths for doing similar things as possible—parallel implementations naturally diverge and then bugs that get fixed in one place remain in another. To the extent that it's practical, I'd love to see all of these very similar functions doing the same thing, only parametrized by little details like which error type to throw. I realize we will probably fall short of that ideal, but I think we should aim for it over hypothetical performance concerns (until performance impacts are actually measured).

Reply:
In that case, we should always use `unifyishRows`.

Reply:
In latest commit, I've updated Kinds to always use `unifyishRows` and guarantee that `KindsDoNotUnify` is the error created in all circumstances.